test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -373,6 +373,12 @@ void UcxContext::progress_io_message()
         UCX_LOG << "could not find connection with id " << conn_id;
     } else {
         UcxConnection *conn = iter->second;
+        if (!conn->is_established()) {
```

Comment:
maybe assert is_am==false?

Reply:
> maybe assert is_am==false?

IMO does not make sense, `_use_am` just checked above

Reply:
> can add `unlikely`?

standard likely/unlikely since C++20, we did not use `ucs` macro until now

Reply:
> > maybe assert is_am==false?
> 
> IMO does not make sense, `_use_am` just checked above

right.. code was folded so didn't notice it